### ISSUE TITLE
Checkbox & Switch currently have multiple labels, which axe errors on

### ISSUE THREAD

**Issue by jpzwarte** [sl-design-system/components#1776]
### Provide a general summary of the issue here

So `<sl-checkbox>` currently fails https://dequeuniversity.com/rules/axe/4.7/form-field-multiple-labels.

As shows on that page, it can be tweaked to not fail.

![Image](https://github.com/user-attachments/assets/b0ddca98-84bc-4f2f-9d7d-cc513bbe89e2)

![Image](https://github.com/user-attachments/assets/3d32189a-68a1-4594-977c-b75137dc7b1e)

### 👤 Your name

Jeroen

### 🧢 Your Product/Team

SLDS
